prefect_aws/s3bucket.py

```diff
@@ -0,0 +1,80 @@
+import io
+from typing import Optional, Union
+from uuid import uuid4
+
+import boto3
+from anyio import to_thread
+from prefect.filesystems import ReadableFileSystem, WritableFileSystem
+
+from prefect_aws import AwsCredentials, MinIOCredentials
+
+
+class S3Bucket(ReadableFileSystem, WritableFileSystem):
+
+    """
+    Block used to store data using S3-compatible object storage like MinIO.
+
+    Args:
+        bucket_name: Name of your bucket.
+        credentials: A block containing your credentials (AwsCredentials or
+        MinIOCredentials).
+        basepath: Used when you don't want to read/write at base level.
+        endpoint_url: Used for non-AWS configuration. When unspecified,
+        defaults to AWS.
+
+    Example:
+        Load stored S3Bucket configuration:
+        ```python
+        from prefect_aws import S3Bucket
+
+        s3bucket_block = S3Bucket.load("BLOCK_NAME")
+        ```
+    """
+
+    # change
+    _logo_url = "https://w7.pngwing.com/pngs/564/59/png-transparent-amazon-com-amazon-s3-amazon-web-services-amazon-simple-queue-service-amazon-glacier-bucket-miscellaneous-data-amazon-dynamodb.png"  # noqa
+    _block_type_name = "S3 Bucket"
+
+    bucket_name: str
+    credentials: Union[MinIOCredentials, AwsCredentials]
+    basepath: Optional[str] = None
+    endpoint_url: Optional[str] = None
+
+    def _get_s3_client(self) -> boto3.client:
+
+        # MinIO
+        if self.endpoint_url:
+            s3_client = boto3.client(
+                service_name="s3",
+                aws_access_key_id=self.credentials.minio_root_user,
+                aws_secret_access_key=self.credentials.minio_root_password.get_secret_value(),  # noqa
+                endpoint_url=self.endpoint_url,
+            )
+
+        # AWS
+        else:
+            s3_client = boto3.client(service_name="s3")
```

Comment:
```suggestion
        s3_client_kwargs  = {}
        if self.endpoint_url:
            # minio
            aws_secret_access_key = self.credentials.minio_root_password
            s3_client_kwargs.update(
                    aws_access_key_id=self.credentials.minio_root_user,
                    aws_secret_access_key=aws_secret_access_key.get_secret_value(),
                    endpoint_url=self.endpoint_url,
             )
        s3_client = boto3.client(service_name="s3", **s3_client_kwargs)
```

Reply:
Although I wonder if we should have a MinIOS3Bucket that inherits from S3Bucket and overrides this _get_s3_client method